c++/src/kj/async-test.c++

```diff
@@ -401,12 +403,24 @@ TEST(Async, Ordering) {
       EXPECT_EQ(4, counter++);
     }).then([&]() {
       EXPECT_EQ(5, counter++);
+      promises[8] = kj::evalLast([&]() {
```

Comment:
So, I found this test to be kind of a fun brainteaser...  I was trying to work out why the evalLast() doc comment says it fires in LIFO order, while the two evalLast() calls in the test appear to be firing in FIFO order.  As far as I can tell, the armLast() code does indeed seem to queue events in LIFO order, and the reason the evalLast() lambdas run in the opposite order is because the intervening promise structures call armLast() in the opposite order.

Is this a demonstration of the doc comment's rule "If the first callback enqueues new events, then latter callbacks will not execute until those events are drained."?  If so, it might be worth a comment in the test itself...  The other comments in the test were much-valued breadcrumbs in puzzling out how the system is supposed to work.  Or perhaps it might be clearer broken out into a separate test?

Also, I'm guessing that LIFO order of simple evalLast() lambdas is also important to guarantee, in which case we might want to also cover it in a test?

Reply:
Erm. TBH now that you mention it my test code appears to be checking for the wrong ordering, and I don't know why it passes.

Reply:
Oh I bet it's because I didn't do `.eagerlyEvaluate(nullptr)` everywhere and so the ordering turned out to be governed by the ordering of the waits on the promise array.

I really don't know why, when I originally wrote this test, I used a promise array rather than a vector or a TaskSet or something. Maybe TaskSet didn't exist yet... I should rewrite the test to avoid confusing myself.

Reply:
Ah, yeah... I saw the lack of ".eagerlyEvaluate", but I assumed it was intentional.

Reply:
Fixed, and rewrote the test to use TaskSet.